doc/api/api_overview.rst

```diff
@@ -0,0 +1,54 @@
+API Overview
+============
+
+Below we describe several common approaches to plotting with Matplotlib.
+
+.. contents::
+
+The pyplot API
+--------------
+
+`matplotlib.pyplot` is a collection of command style functions that make
+Matplotlib work like MATLAB. Each pyplot function makes some change to a
+figure: e.g., creates a figure, creates a plotting area in a figure, plots
+some lines in a plotting area, decorates the plot with labels, etc.
+
+`.pyplot` is mainly intended for interactive plots and simple cases of
+programmatic plot generation.
+
+Further reading:
+
+- The `matplotlib.pyplot` function reference
+- :doc:`/tutorials/introductory/pyplot`
+- :ref:`Pyplot examples <pyplots_examples>`
+
+The object-oriented API
+-----------------------
+
+At its core, Matbplotlib is object-oriented. We recommend directly working
+with the objects, if you need more control and customization of your plots.
+
+In many cases you will create a `.Figure` and one or more
+`~matplotlib.axes.Axes` using `.pyplot.subplots` and from then on only work
+on these objects. However, it's also possible to create `.Figure`\ s
+explicitly (e.g. when including them in GUI applications).
+
+Further reading:
+
+- `matplotlib.axes.Axes` and `matplotlib.figure.Figure` for an overview of
+  plotting functions.
+- Most of the :ref:`examples<api_examples>` use the object-oriented approach.
+
+
+The pylab API
```

Comment:
I'm pretty sure we stopped documenting the pylab API on purpose, with the hope that "if it's not documented, it doesn't exist" (which *clearly* doesn't work in the case of pylab :( ).

Reply:
I think it's better to clearly say: "pylab exists for backward compatibility, but don't use it" than to not mention pylab at all in the docs. Since there are so people who use(d) it and so many examples out there, not mentioning pylab is rather confusing than helpful.

I can make the wording stronger. Is pylab officially deprecated?

Reply:
Define "officially deprecated" :)

matplotlib.pylab doesn't and has never raised a deprecation warning, but I've heard for years now that it was deprecated. It was added in conjecture of the ``ipython --pylab`` option that is also something I heard was deprecated (@Carreau may be able to confirm), but doesn't raise a deprecation warning on the version of ipython I have.

The examples were supposed to be moving away from the pylab interface and to the Object-Oriented one, but I don't believe anyone has been actively working on this aspect of documentation recently.

This is all very vague, and should probably not be considered into the review unless someone else confirms my memories of us intentionally not documenting pylab anymore.

Reply:
"officially deprecated" means there is a public statement, that this API should not be used anymore. 😄.

Don't know if that already exists (not found anything in the docs 😛), but I'm all in for it. I've added a deprecation mark and turned the note into a more scary looking warning.